packages/plugin-assets-retry/src/runtime/index.ts

```diff
@@ -278,7 +282,18 @@ function resourceMonitor(
 
 // @ts-expect-error init is a global function, ignore ts(6133)
 function init(options: PluginAssetsRetryOptions) {
-  const config = Object.assign({}, defaultConfig, options);
+  const config: PluginAssetsRetryOptions = {};
+
+  for (const key in defaultConfig) {
+    // @ts-ignore
```

Comment:
can we make the type correct instead of use `@ts-ignore`..

Reply:
I tried... key is type of string and typescript is not always smart.
<img width="318" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/79413249/19ce75bb-19ee-4961-b145-545422e78457">

Reply:
how about just use object spread

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/4df03862-1020-44d8-9ade-34577a640307)

Reply:
@SoonIter should we change to object spread

Reply:
Honestly speaking, I do not like it.  These compatible codes have 2kb.
<img width="300" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/79413249/edb6a3df-c9e9-416c-9e10-6aa97142c4c7">

Reply:
Understand, this is really not worth